documentation/portlet-support/portlet-07-cdi-support.asciidoc

```diff
@@ -0,0 +1,233 @@
+---
+title: Vaadin Portlet CDI Support
+order: 7
+layout: page
+---
+
+= Vaadin Portlet CDI Support
+
+The Vaadin Portlet CDI (Contexts and Dependency Injection) extension allows development of Portlet 3.0 (JSR 362) portlets with Vaadin components and services as managed beans.
+With the Vaadin Portlet CDI extension, managed beans can be injected into any Vaadin component with the application container managing the initialization and destruction (lifecycle) of beans automatically.
+
+As a developer, when would I want to use CDI for my Vaadin Portlet application?
+The primary use cases for Vaadin Portlet CDI are:
+
+- If the business layer or other parts of your Portlet application is already using CDI, and you want to leverage CDI in your UI layer as well.
+- You want to use portlet scopes, such as `@PortletRequestScoped` and `@PortletSessionScoped`, for portlet-specific lifecycle management of your UI or business objects.
+- You want to access portlet predefined beans, as specified by JSR 362.
+
+[NOTE]
+This tutorial assumes you are building Vaadin portlets for Apache Pluto portal, which uses Tomcat and Weld as the application and CDI containers.
+Other CDI implementations are not supported by Vaadin Portlet CDI.
+
+== How Do I Use the Vaadin CDI Portlet extension?
+
+Taking the CDI extension into use in your Vadin Portlet project requires the following steps:
+
+. Add `vaadin-portlet-cdi` dependency
+. Add an empty `beans.xml` configuration file
+. Use `VaadinCdiPortlet` class as the base class for your portlet
+. Add `@Inject`, scope annotations and other CDI features in your code
+
+The remainder of this chapter goes through each of these steps in detail.
+
+[NOTE]
+The Vaadin CDI Portlet extension differs from the Vaadin CDI add-on (<<../cdi/tutorial-cdi-basic#,Use Vaadin with CDI>>) for servlet applications.
+In particular, the scopes introduced by the latter (`@VaadinServiceScoped`, `@VaadinSessionScoped`, `@UIScoped`, `@RouteScoped`) should not be used for portlets, since the session and service concepts differ slightly, and there is no routing.
+Other than this, the standard CDI features work exactly the same.
+
+[NOTE]
+This chapter is not a tutorial on CDI in general. For resources on CDI, see, visit http://www.cdi-spec.org/
```

Comment:
Should we also point to [Using Vaadin with CDI](https://github.com/vaadin/flow-and-components-documentation/blob/master/documentation/cdi/tutorial-cdi-basic.asciidoc) here also?